src/app/main.cpp

```diff
@@ -303,58 +301,41 @@ QBtCommandLineParameters parseCommandLine()
 }
 
 #if defined(Q_OS_UNIX) || defined(STACKTRACE_WIN)
-void sigintHandler(int)
+void sigNormalHandler(int signum)
 {
-    signal(SIGINT, 0);
-    qDebug("Catching SIGINT, exiting cleanly");
-    qApp->exit();
-}
-
-void sigtermHandler(int)
-{
-    signal(SIGTERM, 0);
-    qDebug("Catching SIGTERM, exiting cleanly");
-    qApp->exit();
-}
-
-void sigsegvHandler(int)
-{
-    signal(SIGABRT, 0);
-    signal(SIGSEGV, 0);
 #if !defined Q_OS_WIN && !defined Q_OS_HAIKU
-    std::cerr << "\n\n*************************************************************\n";
-    std::cerr << "Catching SIGSEGV, please report a bug at http://bug.qbittorrent.org\nand provide the following backtrace:\n";
-    std::cerr << "qBittorrent version: " << VERSION << std::endl;
-    print_stacktrace();
-#else
-#ifdef STACKTRACE_WIN
-    StraceDlg dlg;
-    dlg.setStacktraceString(straceWin::getBacktrace());
-    dlg.exec();
-#endif
-#endif
-    raise(SIGSEGV);
+    const char str1[] = "Catching signal: ";
+    const char *sigName = sys_siglist[signum];
+    const char str2[] = "\nExiting cleanly\n";
+    write(STDERR_FILENO, str1, strlen(str1));
```

Comment:
I'm trying to comply with the table here:
http://man7.org/linux/man-pages/man7/signal.7.html (see Async-signal-safe functions section)